Source/SPDatabaseDocument.m

```diff
@@ -5080,9 +5080,11 @@ - (BOOL)setStateFromConnectionFile:(NSString *)path
 	}
 
 	// Move favourites and history into the data dictionary to pass to setState:
-	[data setObject:[spf objectForKey:SPQueryFavorites] forKey:SPQueryFavorites];
-	[data setObject:[spf objectForKey:SPQueryHistory] forKey:SPQueryHistory];
-	[data setObject:[spf objectForKey:SPContentFilters] forKey:SPContentFilters];
+	// SPQueryHistory is no longer saved to the SPF file, so it was causing an exception here (it was adding nil to the spf dict), skipping out of the method and not connecting
+	// or restoring the query screen content. See commit 96063541
+	if([spf objectForKey:SPQueryFavorites])[data setObject:[spf objectForKey:SPQueryFavorites] forKey:SPQueryFavorites];
```

Comment:
Not a fan of conditionals without `{}`, but I don't care that much. Do we have a style guide for development?

Reply:
There is very similar code at L3577-L3579 and L4896-L4898.
Maybe we want to add the same checks there too?

Reply:
> Not a fan of conditionals without `{}`, but I don't care that much. Do we have a style guide for development?

I'm not really either, I've updated.

Reply:
> There is very similar code at L3577-L3579 and L4896-L4898.
> Maybe we want to add the same checks there too?

I've added guards at L3577-L3579. L4896-L4898 is already guarded, it's doing if-assign-set.